core/core_isolate.rs

```diff
@@ -123,6 +134,7 @@ pub struct CoreIsolateState {
   pub(crate) js_macrotask_cb: Option<v8::Global<v8::Function>>,
   pub(crate) pending_promise_exceptions: HashMap<i32, v8::Global<v8::Value>>,
   pub(crate) js_error_create_fn: Box<JSErrorCreateFn>,
+  pub rust_err_to_json_fn: &'static dyn RustErrToJsonFn,
```

Comment:
This would be unnecessary if every ErrBox contained a "class" string.

Reply:
Well *only* if you specify the error class at every single location where an error is converted to an ErrBox (and there are a lot!). You'd have to opt out of using `.into()` and the ? operator entirely.